src/dht/dht_transaction.h

```diff
@@ -252,7 +252,7 @@ class DhtTransaction {
   };
 
   // Key to uniquely identify a transaction with given per-node transaction id.
-  using key_type = uint64_t;
+  using key_type = std::array<uint8_t, 20>;
```

Comment:
This misses the point of a key, as in it should be a built-in type that is fast to look up.

We don't care if part of the key gives us the whole ip address, we just want it to be unique.

We could make the most significant bit of the ID part indicate ipv4/6, and have collision check for ipv6 addresses.